hax/hax/util.py

```diff
@@ -1954,6 +1971,38 @@ def get_process_current_status(self, status_reported: ObjHealth,
                  node, proc_fid, status_current)
         return status_current
 
+    @repeat_if_fails()
+    def get_process_based_node_state(self, node_fid: Fid) -> str:
+        children = self.kv.kv_get(f'm0conf/nodes/{node_fid}/processes',
+                                  recurse=True)
+        total_processes = 0
+        started_processes = 0
+        for item in children or []:
+            # m0conf/nodes/0x6e00000000000001:0x3/processes/
+            # 0x7200000000000001:0xa:
+            # {"name": "m0_server", "state": "online"}
+            regex = re.compile('^m0conf\\/.*\\/processes\\/([^/]+)$')
+            match_result = re.match(regex, item['Key'])
+            if not match_result:
+                continue
+            p_fid = match_result.group(1)
+
+            p_key = f"processes/{p_fid}"
+            process = self.kv.kv_get(p_key, recurse=False)
+            if not process:
+                continue
+            total_processes += 1
+            state = json.loads(process['Value'])['state']
+            # Checking if status is M0_CONF_HA_PROCESS_STARTED
+            if state == ha_process_events[1]:
+                started_processes += 1
+        if total_processes == started_processes:
```

Comment:
This will not work because if the statement at line#1992 is true then we need to still count that process as offline. Since you are incrementing `total_processes` after #1992, if a process has not started yet, it will not be counted and you may still end up with `total_processes == started_processes` , e.g. consider 2 processes configured, but one has not started, then `total_processes = 1` and possibly `started_processes = 1`, then the function would still report online, but we actually want to report degraded.

Reply:
> we need to still count that process as offline.

okay.. initial starting of the processes scenario, I need to consider along with failure scenarios. I should be counting the total configured processes. Thanks for the explanation.

Reply:
done